File: src/main/java/com/hopital/entities/Rendez_Vous.java
package com.hopital.entities;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;
import java.time.LocalTime;
import java.util.Date;

@Data
@NoArgsConstructor
@RequiredArgsConstructor
@Entity
@Table(name = "Rendez_Vous")
public class Rendez_Vous {

    @Id
    @GeneratedValue
    private long id;
    @NonNull
    private Date daterdv;
    @NonNull
    private LocalTime heurerdv;
    @NonNull
    private boolean confirmed;
    @ManyToOne()
    private Patient patient;
}

File: Hospital-Management-SystemFront/src/app/models/Rendez_vous.ts
export class Rendez_vous{
    id?:number;
    daterdv?:string;
    heurerdv?:string;
    confirmed?:boolean;
    constructor(id?:number, daterdv?:string, heurerdv?:string, confirmed?:boolean){
        this.id=id;
        this.daterdv=daterdv;
        this.heurerdv=heurerdv;
        this.confirmed=confirmed;
    }
}